processing/src/main/java/io/druid/query/select/SelectQueryQueryToolChest.java

```diff
@@ -359,9 +360,19 @@ public Object apply(final Result<SelectResultValue> input)
     }
 
     final Granularity granularity = query.getGranularity();
+    final Iterable<String> filteredPagingKeys = Iterables.filter(
+        paging.keySet(), new Predicate<String>()
+        {
+          @Override
+          public boolean apply(String input)
+          {
+            return input.startsWith(String.format("%s_", dataSource));
```

Comment:
Yeah I grabbed the startsWith {dataSource}_ logic from DataSegmentUtils. Will take another look.